### PR TITLE
SW-2683 Update date utils to support time zones

### PR DESCRIPTION
**Pull request by karthikbtf**
- leveraging luxon
- added tests
- updated with some warnings related fixes

### CODE REVIEW

File: src/utils/date.ts [hunk 1]
```diff
@@ -1,13 +1,89 @@
-const getDateDisplayValue = (date: string | number) => new Date(date).toISOString().split('T')[0];
+import { DateTime } from 'luxon';
 
-export const getTodaysDateFormatted = () => {
-  return getDateDisplayValue(Date.now());
+/**
+ * Helper util that returns default time zone as a fallback.
+ */
+const tz = (timeZoneId?: string) => timeZoneId || 'Etc/UTC';
```
Comment (charlie-tf) [inline src/utils/date.ts:6]: Would it be useful for this function to validate that `timeZoneId` is a valid identifier?

File: src/utils/date.ts [hunk 2]
```diff
@@ -1,13 +1,89 @@
...
+const getDateFromMillis = (date:any , timeZoneId?: string) => {
+  const millis = new Date(date).getTime();
+
+  return DateTime.fromMillis(millis, { zone: tz(timeZoneId) });
+};
+
+/**
+ * Return the display value for a date in ISO 8601 format.
+ */
+const getDisplayValue = (date: any) => {
```
Comment (charlie-tf) [inline src/utils/date.ts:28]: Is `any` the correct type here? Or `Date` or `DateTime`?

File: src/utils/date.ts [hunk 3]
```diff
@@ -1,13 +1,89 @@
...
+/**
+ * Helper util that returns default time zone as a fallback.
+ */
+const tz = (timeZoneId?: string) => timeZoneId || 'Etc/UTC';
+
+/**
+ * Parse a date string with zone implied.
+ * eg. '2023-01-10 in 'America/Los_Angeles'
+ */
+const getDateFromString = (date: any, timeZoneId?: string) => {
```
Comment (charlie-tf) [inline src/utils/date.ts:12]: Should this be `date: string`?

Comment (charlie-tf) [reply src/utils/date.ts:12]: Oh, I see that it is called by `getDate`, which explicitly accepts a `number` as well as a string. Could it be `string | number`?

**Review by charlie-tf — APPROVED** [review]
Few questions mostly about type signatures.

File: src/utils/date.ts (hunk 1 above)
Comment (karthikbtf) [reply src/utils/date.ts:6]: I thought about it, the concept of valid time zones is now within terraware-web (returned from BE). we can iterate on this for sure.

File: src/utils/date.ts (hunk 2 above)
Comment (karthikbtf) [reply src/utils/date.ts:28]: this could be DateTime, let me check

File: src/utils/date.ts (hunk 3 above)
Comment (karthikbtf) [reply src/utils/date.ts:12]: also, i realized these could be breaking changes - going to rename this file and restore the original version. We can use the correct file based on opt-in flag.

File: src/utils/date.ts (hunk 2 above)
Comment (karthikbtf) [reply src/utils/date.ts:28]: yup DateTime, will update.. and the others are string | number

**Review by charlie-tf — APPROVED** [review]
👍

**Review by charlie-tf — APPROVED** [review]
:shipit:

**Pull request merged**